satellite.go

```diff
@@ -0,0 +1,214 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/briandowns/spinner"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+type SatellitePingResponse struct {
```

Comment:
This is a big structure, and the HTTP API does not document this object at all.
If these fields are not required as part of the handling of the ping response,
it's safer to not try to unmarshal them at all.

Reply:
Yeah, I wasn't able to find any documentation for output of this REST API call (https://satellite.sat.engineering.redhat.com/apidoc/v2/ping/ping.en.html). I think it would be worth asking Satellite developers how stable format of this object is.

I would like to emphasize that this ping is only temporary solution that should avoid calling `rhc configure satellite --url <some.url>` against servers that are not satellite servers. More checks we do the better it is. We will replace this solution with calling different REST API call in hopefully near future. I believe that there is low risk that structure of returned object will change soon.

What about reducing this structure to something minimalist like this:

```go
type SatellitePingResponse struct {
	Results struct {
		Katello struct {
			Services struct {
				Candlepin struct {
					Status     string `json:"status"`
					DurationMs string `json:"duration_ms"`
				} `json:"candlepin"`
			} `json:"services"`
			Status string `json:"status"`
		} `json:"katello"`
	} `json:"results"`
}
```

Reply:
Reducing it is important. I think that will reduce the risk of unmarshalling
errors that can turn up with a change to the HTTP response.